### DISCUSSION TITLE
Migrating BSP to new Bitbake sintax

### DISCUSSION THREAD

**Discussion by fabiocabaceira** [OE4T/meta-tegra#1314] [General]
Hello everyone,

Recently, I have been working with the Jetson Orin NX using the Kirkstone branch. I am transforming a BSP layer that only has the Dunfell branch into the Kirkstone branch to make the image work for the Jetson Orin NX.

One of the first problems I encountered was that the bitbake syntax had changed. After researching the Yocto and BitBake documentation, I discovered that Yocto has provided three scripts to help migrate from the old syntax to the new one (convert-overrides.py, convert-variable-renames.py, and convert-srcuri.py).

I have created a bash script attached to this message to make this process a bit easier for me. I am sharing this message to see if I missed anything regarding the migration from the old syntax to the new one and to ask if you have any tips.

Instructions for running the script:

Create the script under layers/scripts for it to work properly.
Example: tegra-demo-distro/layers/scripts$ touch syntax.sh
Run the script under layers/scripts for it to work properly.
Example: tegra-demo-distro/layers/scripts$ bash syntax.sh
[sintax.txt](https://github.com/OE4T/meta-tegra/files/12290861/sintax.txt)

Thank you.

**Comment by quaresmajose** [comment]
Hi Fabio,

You just need to run the [convert-overrides](https://git.yoctoproject.org/poky/tree/scripts/contrib/convert-overrides.py) on a first step because it fixes the most critical bitbake parsing errors.
The other two scripts, `convert-variable-renames.py` and `convert-srcuri.py`, are useful but not critical on a first step.

You can check the conversion seeing the status of the bitbake parsing runing `bitbake -p` this last command will give some hints show what is missing if something is missing.

**Reply by fabiocabaceira** [reply]
Thanks @quaresmajose .
